modules/csg/csg_shape.cpp

```diff
@@ -653,7 +653,7 @@ void CSGShape3D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_meshes"), &CSGShape3D::get_meshes);
 
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "operation", PROPERTY_HINT_ENUM, "Union,Intersection,Subtraction"), "set_operation", "get_operation");
-	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "snap", PROPERTY_HINT_RANGE, "0.0001,1,0.001,suffix:m"), "set_snap", "get_snap");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "snap", PROPERTY_HINT_RANGE, "0.001,1,0.001,suffix:mm"), "set_snap", "get_snap");
```

Comment:
You add a factor 100 so that wouldn't go from meter to millimeter. Was the previous value in decimeters?

Reply:
This is the bugfix for the very small box sample someone made. It increases the precision range. Not sure if this is the best way but it has an effect on that test case.

Reply:
https://github.com/godotengine/godot/files/10949662/CSGBug_2023-03-11.zip I am not able check immediately. Sorry

Reply:
to be clear a 10 cm hole is too much

Reply:
The default snap is `0.001` (m), which is 1 mm. That shouldn't be 10 cm, which would be `0.1`.

Reply:
I removed the `snap * 100` change so we don't break compatibility.

I changed the min value and step to `1e-6` so users can adjust the snap value further down if they need.

Changing the snap to `0.0001` fixes both MRPs from #41140 in my tests. I don't know if that's a good enough resolution for that issue, CC @goatchurchprime.

Reply:
The reasoning makes sense.